src/lem/constrainer.rs

```diff
@@ -44,14 +45,71 @@ impl<F: LurkField> AllocationManager<F> {
             }
         }
     }
+
+    /// Calls `get_or_alloc_num` to allocate tag and hash for a pointer.
+    pub(crate) fn get_or_alloc_ptr<CS: ConstraintSystem<F>>(
+        &mut self,
+        cs: &mut CS,
+        z_ptr: &ZPtr<F>,
+    ) -> Result<AllocatedPtr<F>> {
+        Ok(AllocatedPtr::from_parts(
+            &self.get_or_alloc_num(cs, z_ptr.tag.to_field())?,
+            &self.get_or_alloc_num(cs, z_ptr.hash)?,
```

Comment:
Similar question to above — I guess this function is intended for 'constant ptrs'?